File: frontend/src/types/content.ts
export interface ContentResponse {
    content: string;
    locale: string;
    updated_at: string;
  }
  
export interface HomeContent {
  content: {
    hero: {
      title: string;
      subtitle: string;
      ctaPrimary: string;
      ctaSecondary: string;
    };
    features: {
      title: string;
      subtitle: string;
      items: Array<{
        icon: 'Brain' | 'Sparkles' | 'Zap';
        title: string;
        description: string;
        }>;
      };
    };
  }
  
  export interface ApiResponse<T> {
    data: T;
    message?: string;
  }

File: app/schemas/content.py
from pydantic import BaseModel
from typing import List, Literal

class Hero(BaseModel):
    title: str
    subtitle: str
    ctaPrimary: str
    ctaSecondary: str

class FeatureItem(BaseModel):
    icon: Literal['Brain', 'Sparkles', 'Zap']
    title: str
    description: str

class Features(BaseModel):
    title: str
    subtitle: str
    items: List[FeatureItem]

class HomeContent(BaseModel):   
    hero: Hero
    features: Features

class ContentResponse(BaseModel):
    content: HomeContent
